app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListFragment.kt

```diff
@@ -90,52 +91,48 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
-                // TODO multiselection
-            }
-
-            header.searchViewCard.apply {
-                // TODO filterButton doesn't propagate the event to root, must display it ?
-                searchView.isGone = true
-                searchViewText.isVisible = true
-                filterButton.isEnabled = false
-                root.setOnClickListener {
-                    safeNavigate(ThreadListFragmentDirections.actionThreadListFragmentToSearchFragment())
-                }
+    private fun setupListeners() = with(binding) {
+        // TODO multiselection
+        // openMultiselectButton.setOnClickListener {}
+
+        searchViewCard.apply {
+            // TODO filterButton doesn't propagate the event to root, must display it ?
+            searchView.isGone = true
+            searchViewText.isVisible = true
+            filterButton.isEnabled = false
+            root.setOnClickListener {
+                safeNavigate(ThreadListFragmentDirections.actionThreadListFragmentToSearchFragment())
             }
+        }
 
-            header.userAvatar.setOnClickListener {
-                safeNavigate(ThreadListFragmentDirections.actionThreadListFragmentToSwitchUserFragment())
-            }
+        userAvatar.setOnClickListener {
+            safeNavigate(ThreadListFragmentDirections.actionThreadListFragmentToSwitchUserFragment())
+        }
 
-            newMessageFab.setOnClickListener {
-                safeNavigate(ThreadListFragmentDirections.actionHomeFragmentToNewMessageActivity())
-            }
+        newMessageFab.setOnClickListener {
+            safeNavigate(ThreadListFragmentDirections.actionHomeFragmentToNewMessageActivity())
+        }
 
-            threadsList.addOnScrollListener(object : RecyclerView.OnScrollListener() {
-                override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
-                    super.onScrollStateChanged(recyclerView, newState)
-                    val layoutManager = recyclerView.layoutManager as LinearLayoutManager
-                    if (layoutManager.findFirstCompletelyVisibleItemPosition() == 0 ||
-                        layoutManager.findLastCompletelyVisibleItemPosition() == threadListAdapter.itemCount - 1
-                    ) {
-                        newMessageFab.extend()
-                    } else {
-                        newMessageFab.shrink()
-                    }
+        threadsList.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                val layoutManager = recyclerView.layoutManager as LinearLayoutManager
+                if (layoutManager.findFirstCompletelyVisibleItemPosition() == 0 ||
+                    layoutManager.findLastCompletelyVisibleItemPosition() == threadListAdapter.itemCount - 1
```

Comment:
- replace `threadListAdapter.itemCount - 1` with `lastIndex`
- use `with` for `layoutManager`

Reply:
done